src/interface/mesh_data.hpp

```diff
@@ -118,15 +118,16 @@ class MeshData {
     }
   }
 
-  void Copy(const std::shared_ptr<MeshData<T>> src) {
+  template <typename... Args>
+  void Copy(const std::shared_ptr<MeshData<T>> src, Args &&... args) {
     const int nblocks = src->NumBlocks();
```

Comment:
Shared pointers do not guarantee you you are not pointing to null. It would be safer to make sure that src is valid before calling it's method.